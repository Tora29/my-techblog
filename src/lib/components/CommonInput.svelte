<script lang="ts">
	// コンポーネントの変数
	export let placeholder: string = 'Input Here';
	export let type: 'text' | 'email' | 'number' | 'password' | 'date' | 'tel' = 'text';
	export let componentType: 'input' | 'label' = 'input';
	export let style: 'input-bordered' | 'input-ghost' = 'input-bordered';
	export let color:
		| 'input-primary'
		| 'input-secondary'
		| 'input-accent'
		| 'input-info'
		| 'input-success'
		| 'input-warning'
		| 'input-error' = 'input-primary';
	export let size: 'input-lg' | 'input-md' | 'input-sm' | 'input-xs' = 'input-md';
</script>

<input {type} {placeholder} class="{componentType} {style} {color} {size}" />
