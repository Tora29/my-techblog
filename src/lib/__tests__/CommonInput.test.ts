import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import CommonInput from '$lib/components/CommonInput.svelte';

describe('TextInput Component', () => {
	test('インプットエリアをデフォルトプロパティでレンダリング', () => {
		const { getByPlaceholderText, container } = render(CommonInput);

		// プレイスホルダーのデフォルト値を確認
		expect(getByPlaceholderText('Input Here')).toBeInTheDocument();

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});

	test('インプットエリアをカスタムプロパティでレンダリング', () => {
		const props = {
			placeholder: 'test-placeholder',
			type: 'password',
			componentType: 'label',
			style: 'input-ghost',
			color: 'input-accent',
			size: 'input-lg'
		};

		const { getByPlaceholderText, getByRole, container } = render(CommonInput, {
			props: {
				placeholder: props.placeholder,
				type: props.type as 'number' | 'text' | 'email' | 'password' | 'date' | 'tel',
				componentType: props.componentType as 'input' | 'label',
				style: props.style as 'input-bordered' | 'input-ghost',
				color: props.color as
					| 'input-primary'
					| 'input-secondary'
					| 'input-accent'
					| 'input-info'
					| 'input-success'
					| 'input-warning'
					| 'input-error',
				size: props.size as 'input-lg' | 'input-md' | 'input-sm' | 'input-xs'
			}
		});

		// カスタムプレイスホルダーの値を確認
		expect(getByPlaceholderText(props.placeholder)).toBeInTheDocument();

		// componentTypeが'input'の場合のみtextboxを取得
		if (props.componentType === 'input') {
			const input = getByRole('textbox');
			expect(input).toHaveClass(
				props.type,
				props.placeholder,
				props.componentType,
				props.style,
				props.color,
				props.size
			);
		}

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
