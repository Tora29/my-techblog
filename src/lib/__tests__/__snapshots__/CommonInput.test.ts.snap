// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextInput Component > インプットエリアをカスタムプロパティでレンダリング 1`] = `
<div>
  <input
    class="label input-ghost input-accent input-lg"
    placeholder="test-placeholder"
    type="password"
  />
</div>
`;

exports[`TextInput Component > インプットエリアをデフォルトプロパティでレンダリング 1`] = `
<div>
  <input
    class="input input-bordered input-primary input-md"
    placeholder="Input Here"
    type="text"
  />
</div>
`;
